core/state_processor.go

```diff
@@ -150,12 +155,25 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 					}
 
 					if acc.HasStorage() {
+					stIt, err := statedb.Snaps().StorageIterator(mpt.Hash(), accIt.Hash(), common.Hash{})
+					if err != nil {
+						return nil, nil, 0, err
+					}
 						for ; stIt.Next() && count < maxMovedCount; count++ {
+						var (
+							value     []byte   // slot value after RLP decoding
+							safeValue [32]byte // 32-byte aligned value
+						)
+						if err := rlp.DecodeBytes(stIt.Slot(), &value); err != nil {
```

Comment:
in the MPT, the data is serialized using RLP. The verkle tree does not do this, so to keep it consistent, deserialize here.